### PR TITLE
indexer: add State method

### PR DESCRIPTION
**Pull request by hdonnay**
This commit also adds openapi documentation for it.

### CODE REVIEW

**Comment by hdonnay** [comment]
This need some additional commits, my open finger got itchy.

**Comment by hdonnay** [comment]
OK, ready

File: indexer/httpclient.go [hunk 1]
```diff
@@ -47,19 +47,19 @@ func (s *httpClient) Index(ctx context.Context, manifest *claircore.Manifest) (*
 		return nil, &clairerror.ErrBadManifest{err}
 	}
 
-	url := url.URL{
-		Scheme: s.addr.Scheme,
-		Host:   s.addr.Hostname(),
-		Path:   IndexAPIPath,
+	u, err := s.addr.Parse(IndexAPIPath)
```
Comment (ldelossa) [inline indexer/httpclient.go:50]: I don't see anything in GoDoc about the Parse method adding a Path field to the existing URL. This is how it works here?

**Review by ldelossa — APPROVED** [review]

File: indexer/httpclient.go (hunk 1 above)
Comment (hdonnay) [reply indexer/httpclient.go:50]: Yeah, it parses it in the context of the *net.URL object: [link](https://pkg.go.dev/net/url?tab=doc#URL.Parse). The previous code would strip port numbers, for example.

**Comment by ldelossa** [comment]
Nice! Didnt see that before. Will make use of that.

**Pull request merged**